### ISSUE TITLE
Diagnostic from Perl code

### ISSUE THREAD

**Issue by kjeldbeg** [SystemRDL/systemrdl-compiler#108]
I have some Perl code in the RDL. Occasionally the Perl code encounters a problem that should be reported as an error. It is not possible to do a print. What is the recommended way to do it?

**Comment by amykyta3** [comment]
You can force an error from the perl preprocessor using the `die` keyword:
```perl
<%
    if(1) {
        die "My error happened";
    }
%>
```
This will throw an exception which causes the preprocessor to dump the output of STDERR:
```
perl_error.rdl: fatal: Encountered a Perl syntax error while executing embedded Perl preprocessor commands:
My error happened at (eval 9) line 4, <STDIN> line 7.
```

Currently, the line numbers reported by Perl are pretty meaningless and should be ignored. I plan to address that at some point.
